### PR TITLE
[configure] [coqide] Put coqide config in its own file.

### PR DESCRIPTION
**Pull request by ejgallego**
Follow up of #16512 , now coqide has its own configure script.

cc #7149

### CODE REVIEW

File: ide/coqide/dune [hunk 1]
```diff
@@ -23,19 +23,24 @@
  (files (idetop.bc as coqidetop.byte)))
 
 ; IDE Client, we may want to add the macos_prehook.ml conditionally.
+(library
+ (name coqide_config)
+ (wrapped false)
+ (modules config))
```
Comment (SkySkimmer) [inline ide/coqide/dune:29]: I don't like having a generic name like `config` not wrapped

**Review by SkySkimmer — COMMENTED** [review]
seems broken as dune doesn't know how to generate config.ml

File: ide/coqide/dune (hunk 1 above)
Comment (ejgallego) [reply ide/coqide/dune:29]: Library is private tho, but we can wrap.

**Comment by ejgallego** [comment]
> seems broken as dune doesn't know how to generate config.ml

Ups.

File: ide/coqide/dune (hunk 1 above)
Comment (ejgallego) [reply ide/coqide/dune:29]: Library gone.

**Review by Alizter — APPROVED** [review]

**Comment by Alizter** [comment]
@coqbot merge now

**Comment by Alizter** [comment]
@coqbot merge now

**Pull request merged**